packages/runtime-dom/src/modules/attrs.ts

```diff
@@ -80,13 +80,12 @@ export function compatCoerceAttr(
   } else if (
     value === false &&
     !isSpecialBooleanAttr(key) &&
-    compatUtils.isCompatEnabled(DeprecationTypes.ATTR_FALSE_VALUE, instance)
-  ) {
-    compatUtils.warnDeprecation(
+    compatUtils.softAssertCompatEnabled(
```

Comment:
This change seems not to be proper.
see https://github.com/vuejs/core/commit/04729ba2163d840f0ca7866bc964696eb5557804
see https://github.com/vuejs/core/pull/11126#issuecomment-2167586454

Reply:
The changes in `compatConfig.ts`
```ts 
if (config === false || config === 'suppress-warning') {
   return
}
``` 
fix the this PR related issue and #11126 also, so this change is just revert the some slightly cumbersome logic of 04729ba.
Moreover, it is possible to see that both the original test and the new test have passed.